File: javaDesde0/ejercicios1/ProyectoP1/Taxi.java
public class Taxi{ //El nombre de la clase
    private String ciudad; // ciudad de cada objeto taxi
    private String matricula; // Matricula de cada objeto taxi
    private String distrito; // Distrito asiganado a cada objeto taxi
    private int tipoMotor; // tipo de motor asignado a cada objeto taxi. 0 = desconocido, 1 = gasolina, 2 = disel
 
    
    //Constructor: cuando se cree un objeto taxi se ejecutara el codigo que incluyamos en el constructor
    public Taxi(){
        
        ciudad = "Guatemala";
        matricula = "Foc2813";
        distrito = "Desconocido";
        tipoMotor = 2;
        
    } //Cierre del constructor ... el codigo continua
    
    public Taxi(String valorMatricula, String valorDistrito, int valorTipoMotor){
    
        ciudad = "Guatemala";
        matricula = valorMatricula;
        distrito = valorDistrito;
        tipoMotor = valorTipoMotor;
        
    }
    
    //Metodo para establecer la matricula de un taxi
    public void setMatricula(String valorMatricula){
        
        matricula = valorMatricula;// La matricula del objeto taxi adopta el valor que contenga valorMatricula
    
    }//Cierre del metodo
    
    public void setDistrito(String valorDistrito){
        
        distrito  = "Distrito " + valorDistrito;// el valor distrito del objeto taxi odopta el valor indicado
        
    }
    
    //Metodo para establecer el distrito de un taxi
    public void setTipoMotor(int valorTipoMotor){
    
        tipoMotor = valorTipoMotor; //El tipoMotor del objeto taxi adopta el valor que contenga valorTipoMotor
    
    } // cierre metodo
    
    //Metodo para obtener la matricula del objeto taxi
    public String getMatricula() {return matricula;} //Cierre del metodo
    
    //Metodo para obtener el distrito del objeto taxi
    public String getDistrito() {return distrito;}//Cierre del metodo
    
    //Metodo para obtener el tipo de motor del objeto taxi
    public int getTipoMotor() {return tipoMotor;}//Cierre del metodo
    
}//cierre de la clase

File: ejercicios_Python_2/conversionPython/clases1/Taxi.py
class Taxi:

    ciudad: str = None
    matricula: str = None
    distrito: str = None
    tipoMotor: int = 0

    def __init__(self,ciudad:str,matricula:str,distrito:str,tipoMotor:int):
        self.ciudad = ciudad
        self.matricula = matricula
        self.distrito = distrito
        self.tipoMotor = tipoMotor

    def Taxi(self,ciudad:str, matricula: str, distrito:str, tipoMotor:int ):
        ciudad = "guatemala"
        self.ciudad = ciudad
        self.matricula = matricula
        self.distrito = distrito
        self.tipoMotor = tipoMotor

    def setmatricula(self, matricula: str):
        matricula = input("Ingresa la matricula:")
        self.matricula = matricula

    def setDistrito(self, distrito: str):
        distrito = input("Ingresa el distrito:")
        self.distrito = distrito

    def setTipoMotor(self, tipoMotor: int):

        print("Ingrese el numero 1 o 2")
        print("1) motor a disel")
        print("2) motor electrico")
        tipoMotor = int(input("Ingrese el motor:"))

        if tipoMotor == 1:
            tipoMotor: str = "motor a disel"
        elif tipoMotor == 2:
            tipoMotor: str = "motor electrico"
        else:
            print("debe elegir entre 1 y 2")

        self.tipoMotor = tipoMotor

    


# taxi = Taxi(None,None,None,0)
# taxi.Taxi(taxi.ciudad,taxi.matricula,taxi.distrito,taxi.tipoMotor)

# taxi.setmatricula(taxi.matricula)
# taxi.setDistrito(taxi.distrito)
# taxi.setTipoMotor(taxi.tipoMotor)

# print(taxi.matricula)
# print(taxi.distrito)
# print(taxi.tipoMotor)
